frontend/web/components/App.js

```diff
@@ -140,25 +97,15 @@ const App = class extends Component {
 
   componentDidUpdate(prevProps) {
     if (prevProps.location.pathname !== this.props.location.pathname) {
-      const newProjectId = this.getProjectId(this.props)
-      if (this.state.projectId !== newProjectId && !!newProjectId) {
-        this.state.projectId = newProjectId
-        AppActions.getProject(this.state.projectId)
+      const newProjectId = this.props.projectId
+      if (prevProps.projectId !== newProjectId && !!newProjectId) {
+        AppActions.getProject(this.props.projectId)
       }
 
       if (isMobile) {
         this.setState({ asideIsVisible: false })
       }
       this.hideMobileNav()
-      AsyncStorage.getItem('lastEnv').then((res) => {
```

Comment:
using lastEnvironment / last project was a bad pattern, everything should just be based on the url